client/banner.js

```diff
@@ -1,19 +1,81 @@
 /*global wp */
+/*global ajaxurl */
+/*global wcs_nux_notice */
 import jQuery from 'jquery';
 import '../assets/stylesheets/banner.scss';
 
 jQuery( document ).ready( ( $ ) => {
-	$( '.wcc-admin-notice .wcc-install-jetpack' ).one( 'click', function() {
+	$( '.woocommerce-services__connect-jetpack' ).one( 'click', function( event ) {
+		event.preventDefault();
 		const btn = $( this );
+		btn.addClass( 'disabled' );
 
-		btn.html( wp.updates.l10n.installing )
-			.addClass( 'disabled' );
+		installStep()
+			.then( activateStep )
+			.then( connectStep )
+			.fail( function( error ) {
+				let errorMessage = error;
+				if ( ! error ) {
+					errorMessage = wcs_nux_notice.translations.defaultError;
+				}
+				if ( error && error.install && 'plugin' === error.install ) {
+					// plugin install error
+					errorMessage = wcs_nux_notice.translations.installError;
+				}
+				$( '<p/>', {
+					// eslint-disable-next-line quote-props
+					'class': 'woocommerce-services__jetpack-install-error-message',
+					text: errorMessage,
+				} ).insertAfter( btn );
+				btn.remove();
+			} );
 
-		wp.updates.installPlugin( { slug: 'jetpack' } )
-			.then( ( response ) => window.location = response.activateUrl || window.location.href )
-			.fail( () =>
-				btn.parent().addClass( 'error' )
-					.html( btn.data( 'error-message' ) )
-			);
+		function installStep() {
+			if ( 'uninstalled' === wcs_nux_notice.initial_install_status ) {
+				return $.when()
+					.then( function() {
+						btn.html( wp.updates.l10n.installing );
+						return wp.updates.installPlugin( { slug: 'jetpack' } );
+					} );
+			}
+			return $.Deferred().resolve();
+		}
+
+		function activateStep() {
+			if ( 'installed' === wcs_nux_notice.initial_install_status ||
+				'uninstalled' === wcs_nux_notice.initial_install_status
```

Comment:
Hmm, I think this is probably not written in the best possible way. Here we're checking if we want to go through the activation step at all, not whether it will be the first step.

We do want to go through activation step (possibly after the installation step) in these cases:
1. When Jetpack is not yet installed (condition 1)
2. After installing Jetpack from scratch (condition 2)

Maybe there's a better way to indicate that's what's happening. Maybe using a variable like `shouldDoActivationStep`?